app/models/world_location.rb

```diff
@@ -13,20 +13,21 @@ def self.reset_cache
 
   def self.all
     cache_fetch("all") do
-      world_locations = Services.worldwide_api.world_locations.with_subsequent_pages.map do |l|
-        new(l) if l.format == "World location" && l.details && l.details.slug.present?
+      world_locations = Services.worldwide_api.world_locations.with_subsequent_pages.map do |response|
+        location = Location.new(response.to_hash)
+        if location.format == "World location" && location.details &&
+            location.slug.present?
+          self.new(location)
+        end
       end
       world_locations.compact
     end
   end
 
   def self.find(location_slug)
     cache_fetch("find_#{location_slug}") do
```

Comment:
@ikennaokpala could you please if we need to cache information locally. It seems that every adapter is already doing it? If so, could you prepare another commit to remove it?